accern_xyme/accern_xyme.py

```diff
@@ -1620,18 +1687,35 @@ def append_jsons(self, jsons: List[Any]) -> 'JSONBlobHandle':
 
 
 class ComputationHandle:
-    def __init__(self, pipeline: PipelineHandle, data_id: str) -> None:
+    def __init__(
+            self,
+            pipeline: PipelineHandle,
+            data_id: str,
+            get_dyn_error: Callable[[], Optional[str]],
+            set_dyn_error: Callable[[str], None]) -> None:
         self._pipeline = pipeline
         self._data_id = data_id
         self._value: Optional[ByteResponse] = None
+        self._get_dyn_error = get_dyn_error
+        self._set_dyn_error = set_dyn_error
 
     def has_fetched(self) -> bool:
         return self._value is not None
 
     def get(self) -> ByteResponse:
-        if self._value is None:
-            self._value = self._pipeline.get_dynamic_result(self._data_id)
-        return self._value
+        maybe_error = self._get_dyn_error()
```

Comment:
if it has a valid value we need to return the valid value -- also, we need to call get_dynamic_result to clear it from redis. it's just that when reporting the error we swap it out with dyn_error